### DISCUSSION TITLE
Lightning sorceress

### DISCUSSION THREAD

**Discussion by murilobc** [isid0re/SoloLeveling#107] [General]
Is there any chance that you can create a Lightning Sorceress script?

Thanks for the excellent work btw.

**Comment by isid0re** [comment]
Yes, with the kind assistance from some users of my script, we are working towards optimizing the leveling builds as well as changing the way the finalbuild respec is called. This way the bot will be able to respec to a light sorc after having infinity, instead of at defined character level. 

Previously I held off from including final light builds because of the immunes. So be patient it will come. And thank you for

**Reply by ddarbs** [reply]
I slammed this together after buying an infinity, and seems to work alright. Need to add ChainL under globals though

/**
 *    @filename   Sorceress.ChainL.js
 *    @desc      Sorceress end-game Light build
 */

var build = {
	caster: true,
	skillstab: 9, //lightning?
	wantedskills: [49, 53, 63], // lightning, chain lightning, lightning mastery
	usefulskills: [48, 57, 42], // nova, thunderstorm,  static
	stats: [
		["energy", 50], ["strength", 48], ["vitality", 165], ["strength", 61], ["vitality", 252], ["strength", 84], ["dexterity", "block"], ["vitality", "all"]
	],
	skills: [
		[37, 1], //warmth 37
		[38, 1], // charged bolt
		[49, 1], // lightning
		[53, 1], //chain lightning 53
		[42, 1], // Static
		[48, 1], //nova 48	
		[57, 1], //thunderstorm 57
		[43, 1], // telekenisis
		[54, 1], // teleport
		[58, 1], //energy shield 58
		[63, 1, false], //lightning mastery, 63 false
		[40, 1], // Frozen Armor
		[39, 1], //ice bolt 39
		[45, 1], //ice blast 45
		[50, 1], //shiver armor 50
		[58, 7], //energy shield to 7  58
		[49, 20], //max lightning 49
		[38, 20], //max charged bolt 38
		[63, 20], //max lightning mastery 63
		[53, 20], //max chain lightning 53
		[48, 20], //max nova 48
	]
};

//	/d2bs/kolbot/libs/config/Builds/Sorceress.ChainL.js

/**
*
* Instructions:	See /d2bs/kolbot/libs/config/Builds/README.txt
*
* Skill IDs:	See /d2bs/kolbot/sdk/skills.txt for a list of skill IDs.
*
* Stat IDs:
*
* 	Strength	= 0
* 	Energy		= 1
* 	Dexterity	= 2
* 	Vitality	= 3
*
*/
js_strict(true);

if (!isIncluded("common/Cubing.js")) {
	include("common/Cubing.js");
}

if (!isIncluded("common/Prototypes.js")) {
	include("common/Prototypes.js");
}

if (!isIncluded("common/Runewords.js")) {
	include("common/Runewords.js");
}

if (!isIncluded("common/Town.js")) {
	include("common/Town.js");
}

var AutoBuildTemplate = {

	1:	{
		//SkillPoints: [-1],								// This doesn't matter. We don't have skill points to spend at lvl 1
		//StatPoints: [-1,-1,-1,-1,-1],						// This doesn't matter. We don't have stat points to spend at lvl 1
		Update: function () {
			Config.AttackSkill = [-1, 49, 42, 53, 42, -1, -1];
			Config.LowManaSkill = [42, 0];
			//Config.SkipImmune = ["fire and cold"];
		}
	},

	2:	{
		SkillPoints: [-1],
		StatPoints: [-1, -1, -1, -1, -1],
		Update: function () {
			Config.AttackSkill = [-1, 49, 42, 53, 42, -1, -1];


		}
	},

	3:	{
		SkillPoints: [-1],
		StatPoints: [-1, -1, -1, -1, -1],
		Update: function () {
			Config.AttackSkill = [-1, 49, 42, 53, 42, -1, -1];

		}
	},

	4:	{
		SkillPoints: [-1],
		StatPoints: [-1, -1, -1, -1, -1],
		Update: function () {
			Config.AttackSkill = [-1, 49, 42, 53, 42, -1, -1];

		}
	},

	5:	{
		SkillPoints: [-1],
		StatPoints: [-1, -1, -1, -1, -1],
		Update: function () {
			Config.AttackSkill = [-1, 49, 42, 53, 42, -1, -1];

		}
	},

	6:	{
		SkillPoints: [-1],
		StatPoints: [-1, -1, -1, -1, -1],
		Update: function () {
			Config.AttackSkill = [-1, 49, 42, 53, 42, -1, -1];

		}
	},

	7:	{
		SkillPoints: [-1],
		StatPoints: [-1, -1, -1, -1, -1],
		Update: function () {
			Config.AttackSkill = [-1, 49, 42, 53, 42, -1, -1];

		}
	},

	8:	{
		SkillPoints: [-1],
		StatPoints: [-1, -1, -1, -1, -1],
		Update: function () {
			Config.AttackSkill = [-1, 49, 42, 53, 42, -1, -1];

		}
	},

	9:	{
		SkillPoints: [-1],
		StatPoints: [-1, -1, -1, -1, -1],
		Update: function () {
			Config.AttackSkill = [-1, 49, 42, 53, 42, -1, -1];

		}
	},

	10:	{
		SkillPoints: [-1],
		StatPoints: [-1, -1, -1, -1, -1],
		Update: function () {
			Config.AttackSkill = [-1, 49, 42, 53, 42, -1, -1];

		}
	},

	11:	{
		SkillPoints: [-1],
		StatPoints: [-1, -1, -1, -1, -1],
		Update: function () {
			Config.AttackSkill = [-1, 49, 42, 53, 42, -1, -1];

		}
	},

	12:	{
		SkillPoints: [-1],
		StatPoints: [-1, -1, -1, -1, -1],
		Update: function () {
			Config.AttackSkill = [-1, 49, 42, 53, 42, -1, -1];

		}
	},

	13:	{
		SkillPoints: [-1],
		StatPoints: [-1, -1, -1, -1, -1],
		Update: function () {
			Config.AttackSkill = [-1, 49, 42, 53, 42, -1, -1];

		}
	},

	14:	{
		SkillPoints: [-1],
		StatPoints: [-1, -1, -1, -1, -1],
		Update: function () {
			Config.AttackSkill = [-1, 49, 42, 53, 42, -1, -1];

		}
	},

	15:	{
		SkillPoints: [-1],
		StatPoints: [-1, -1, -1, -1, -1],
		Update: function () {
			Config.AttackSkill = [-1, 49, 42, 53, 42, -1, -1];

		}
	},

	16:	{
		SkillPoints: [-1],
		StatPoints: [-1, -1, -1, -1, -1],
		Update: function () {
			Config.AttackSkill = [-1, 49, 42, 53, 42, -1, -1];

		}
	},

	17:	{
		SkillPoints: [-1],
		StatPoints: [-1, -1, -1, -1, -1],
		Update: function () {
			Config.AttackSkill = [-1, 49, 42, 53, 42, -1, -1];

		}
	},

	18:	{
		SkillPoints: [-1],
		StatPoints: [-1, -1, -1, -1, -1],
		Update: function () {
			Config.AttackSkill = [-1, 49, 42, 53, 42, -1, -1];

		}
	},

	19:	{
		SkillPoints: [-1],
		StatPoints: [-1, -1, -1, -1, -1],
		Update: function () {
			Config.AttackSkill = [-1, 49, 42, 53, 42, -1, -1];

		}
	},

	20:	{
		SkillPoints: [-1],
		StatPoints: [-1, -1, -1, -1, -1],
		Update: function () {
			Config.AttackSkill = [-1, 49, 42, 53, 42, -1, -1];

		}
	},

	21:	{
		SkillPoints: [-1],
		StatPoints: [-1, -1, -1, -1, -1],
		Update: function () {
			Config.AttackSkill = [-1, 49, 42, 53, 42, -1, -1];

		}
	},

	22:	{
		SkillPoints: [-1],
		StatPoints: [-1, -1, -1, -1, -1],
		Update: function () {
			Config.AttackSkill = [-1, 49, 42, 53, 42, -1, -1];

		}
	},

	23:	{
		SkillPoints: [-1],
		StatPoints: [-1, -1, -1, -1, -1],
		Update: function () {
			Config.AttackSkill = [-1, 49, 42, 53, 42, -1, -1];

		}
	},

	24:	{
		SkillPoints: [-1],
		StatPoints: [-1, -1, -1, -1, -1],
		Update: function () {
			Config.AttackSkill = [-1, 49, 42, 53, 42, -1, -1];

		}
	},

	25:	{
		SkillPoints: [-1],
		StatPoints: [-1, -1, -1, -1, -1],
		Update: function () {
			Config.AttackSkill = [-1, 49, 42, 53, 42, -1, -1];

		}
	},

	26:	{
		SkillPoints: [-1],
		StatPoints: [-1, -1, -1, -1, -1],
		Update: function () {
			Config.AttackSkill = [-1, 49, 42, 53, 42, -1, -1];

		}
	},

	27:	{
		SkillPoints: [-1],
		StatPoints: [-1, -1, -1, -1, -1],
		Update: function () {
			Config.AttackSkill = [-1, 49, 42, 53, 42, -1, -1];

		}
	},

	28:	{
		SkillPoints: [-1],
		StatPoints: [-1, -1, -1, -1, -1],
		Update: function () {
			Config.AttackSkill = [-1, 49, 42, 53, 42, -1, -1];

		}
	},

	29:	{
		SkillPoints: [-1],
		StatPoints: [-1, -1, -1, -1, -1],
		Update: function () {
			Config.AttackSkill = [-1, 49, 42, 53, 42, -1, -1];

		}
	},

	30:	{
		SkillPoints: [-1],
		StatPoints: [-1, -1, -1, -1, -1],
		Update: function () {
			Config.AttackSkill = [-1, 49, 42, 53, 42, -1, -1];

		}
	},

	31:	{
		SkillPoints: [-1],
		StatPoints: [-1, -1, -1, -1, -1],
		Update: function () {
			Config.AttackSkill = [-1, 49, 42, 53, 42, -1, -1];

		}
	},

	32:	{
		SkillPoints: [-1],
		StatPoints: [-1, -1, -1, -1, -1],
		Update: function () {
			Config.AttackSkill = [-1, 49, 42, 53, 42, -1, -1];

		}
	},

	33:	{
		SkillPoints: [-1],
		StatPoints: [-1, -1, -1, -1, -1],
		Update: function () {
			Config.AttackSkill = [-1, 49, 42, 53, 42, -1, -1];

		}
	},

	34:	{
		SkillPoints: [-1],
		StatPoints: [-1, -1, -1, -1, -1],
		Update: function () {
			Config.AttackSkill = [-1, 49, 42, 53, 42, -1, -1];

		}
	},

	35:	{
		SkillPoints: [-1],
		StatPoints: [-1, -1, -1, -1, -1],
		Update: function () {
			Config.AttackSkill = [-1, 49, 42, 53, 42, -1, -1];

		}
	},

	36:	{
		SkillPoints: [-1],
		StatPoints: [-1, -1, -1, -1, -1],
		Update: function () {
			Config.AttackSkill = [-1, 49, 42, 53, 42, -1, -1];

		}
	},

	37:	{
		SkillPoints: [-1],
		StatPoints: [-1, -1, -1, -1, -1],
		Update: function () {
			Config.AttackSkill = [-1, 49, 42, 53, 42, -1, -1];

		}
	},

	38:	{
		SkillPoints: [-1],
		StatPoints: [-1, -1, -1, -1, -1],
		Update: function () {
			Config.AttackSkill = [-1, 49, 42, 53, 42, -1, -1];

		}
	},

	39:	{
		SkillPoints: [-1],
		StatPoints: [-1, -1, -1, -1, -1],
		Update: function () {
			Config.AttackSkill = [-1, 49, 42, 53, 42, -1, -1];

		}
	},

	40:	{
		SkillPoints: [-1],
		StatPoints: [-1, -1, -1, -1, -1],
		Update: function () {
			Config.AttackSkill = [-1, 49, 42, 53, 42, -1, -1];

		}
	},

	41:	{
		SkillPoints: [-1],
		StatPoints: [-1, -1, -1, -1, -1],
		Update: function () {
			Config.AttackSkill = [-1, 49, 42, 53, 42, -1, -1];

		}
	},

	42:	{
		SkillPoints: [-1],
		StatPoints: [-1, -1, -1, -1, -1],
		Update: function () {
			Config.AttackSkill = [-1, 49, 42, 53, 42, -1, -1];

		}
	},

	43:	{
		SkillPoints: [-1],
		StatPoints: [-1, -1, -1, -1, -1],
		Update: function () {
			Config.AttackSkill = [-1, 49, 42, 53, 42, -1, -1];

		}
	},

	44:	{
		SkillPoints: [-1],
		StatPoints: [-1, -1, -1, -1, -1],
		Update: function () {
			Config.AttackSkill = [-1, 49, 42, 53, 42, -1, -1];

		}
	},

	45:	{
		SkillPoints: [-1],
		StatPoints: [-1, -1, -1, -1, -1],
		Update: function () {
			Config.AttackSkill = [-1, 49, 42, 53, 42, -1, -1];

		}
	},

	46:	{
		SkillPoints: [-1],
		StatPoints: [-1, -1, -1, -1, -1],
		Update: function () {
			Config.AttackSkill = [-1, 49, 42, 53, 42, -1, -1];

		}
	},

	47:	{
		SkillPoints: [-1],
		StatPoints: [-1, -1, -1, -1, -1],
		Update: function () {
			Config.AttackSkill = [-1, 49, 42, 53, 42, -1, -1];

		}
	},

	48:	{
		SkillPoints: [-1],
		StatPoints: [-1, -1, -1, -1, -1],
		Update: function () {
			Config.AttackSkill = [-1, 49, 42, 53, 42, -1, -1];

		}
	},

	49:	{
		SkillPoints: [-1],
		StatPoints: [-1, -1, -1, -1, -1],
		Update: function () {
			Config.AttackSkill = [-1, 49, 42, 53, 42, -1, -1];

		}
	},

	50:	{
		SkillPoints: [-1],
		StatPoints: [-1, -1, -1, -1, -1],
		Update: function () {
			Config.AttackSkill = [-1, 49, 42, 53, 42, -1, -1];

		}
	},

	51:	{
		SkillPoints: [-1],
		StatPoints: [-1, -1, -1, -1, -1],
		Update: function () {
			Config.AttackSkill = [-1, 49, 42, 53, 42, -1, -1];

		}
	},

	52:	{
		SkillPoints: [-1],
		StatPoints: [-1, -1, -1, -1, -1],
		Update: function () {
			Config.AttackSkill = [-1, 49, 42, 53, 42, -1, -1];

		}
	},

	53:	{
		SkillPoints: [-1],
		StatPoints: [-1, -1, -1, -1, -1],
		Update: function () {
			Config.AttackSkill = [-1, 49, 42, 53, 42, -1, -1];

		}
	},

	54:	{
		SkillPoints: [-1],
		StatPoints: [-1, -1, -1, -1, -1],
		Update: function () {
			Config.AttackSkill = [-1, 49, 42, 53, 42, -1, -1];

		}
	},

	55:	{
		SkillPoints: [-1],
		StatPoints: [-1, -1, -1, -1, -1],
		Update: function () {
			Config.AttackSkill = [-1, 49, 42, 53, 42, -1, -1];

		}
	},

	56:	{
		SkillPoints: [-1],
		StatPoints: [-1, -1, -1, -1, -1],
		Update: function () {
			Config.AttackSkill = [-1, 49, 42, 53, 42, -1, -1];

		}
	},

	57:	{
		SkillPoints: [-1],
		StatPoints: [-1, -1, -1, -1, -1],
		Update: function () {
			Config.AttackSkill = [-1, 49, 42, 53, 42, -1, -1];

		}
	},

	58:	{
		SkillPoints: [-1],
		StatPoints: [-1, -1, -1, -1, -1],
		Update: function () {
			Config.AttackSkill = [-1, 49, 42, 53, 42, -1, -1];

		}
	},

	59:	{
		SkillPoints: [-1],
		StatPoints: [-1, -1, -1, -1, -1],
		Update: function () {
			Config.AttackSkill = [-1, 49, 42, 53, 42, -1, -1];

		}
	},

	60:	{
		SkillPoints: [-1],
		StatPoints: [-1, -1, -1, -1, -1],
		Update: function () {
			Config.AttackSkill = [-1, 49, 42, 53, 42, -1, -1];

		}
	},

	61:	{
		SkillPoints: [-1],
		StatPoints: [-1, -1, -1, -1, -1],
		Update: function () {
			Config.AttackSkill = [-1, 49, 42, 53, 42, -1, -1];

		}
	},

	62:	{
		SkillPoints: [-1],
		StatPoints: [-1, -1, -1, -1, -1],
		Update: function () {
			Config.AttackSkill = [-1, 49, 42, 53, 42, -1, -1];

		}
	},

	63:	{
		SkillPoints: [-1],
		StatPoints: [-1, -1, -1, -1, -1],
		Update: function () {
			Config.AttackSkill = [-1, 49, 42, 53, 42, -1, -1];

		}
	},

	64:	{
		SkillPoints: [-1],
		StatPoints: [-1, -1, -1, -1, -1],
		Update: function () {
			Config.AttackSkill = [-1, 49, 42, 53, 42, -1, -1];

		}
	},

	65:	{
		SkillPoints: [-1],
		StatPoints: [-1, -1, -1, -1, -1],
		Update: function () {
			Config.AttackSkill = [-1, 49, 42, 53, 42, -1, -1];


		}
	},

	66:	{
		SkillPoints: [-1],
		StatPoints: [-1, -1, -1, -1, -1],
		Update: function () {
			Config.AttackSkill = [-1, 49, 42, 53, 42, -1, -1];


		}
	},

	67:	{
		SkillPoints: [-1],
		StatPoints: [-1, -1, -1, -1, -1],
		Update: function () {
			Config.AttackSkill = [-1, 49, 42, 53, 42, -1, -1];


		}
	},

	68:	{
		SkillPoints: [-1],
		StatPoints: [-1, -1, -1, -1, -1],
		Update: function () {
			Config.AttackSkill = [-1, 49, 42, 53, 42, -1, -1];


		}
	},

	69:	{
		SkillPoints: [-1],
		StatPoints: [-1, -1, -1, -1, -1],
		Update: function () {
			Config.AttackSkill = [-1, 49, 42, 53, 42, -1, -1];


		}
	},

	70:	{
		SkillPoints: [-1],
		StatPoints: [-1, -1, -1, -1, -1],
		Update: function () {
			Config.AttackSkill = [-1, 49, 42, 53, 42, -1, -1];


		}
	},

	71:	{
		SkillPoints: [-1],
		StatPoints: [-1, -1, -1, -1, -1],
		Update: function () {
			Config.AttackSkill = [-1, 49, 42, 53, 42, -1, -1];


		}
	},

	72:	{
		SkillPoints: [-1],
		StatPoints: [-1, -1, -1, -1, -1],
		Update: function () {
			Config.AttackSkill = [-1, 49, 42, 53, 42, -1, -1];


		}
	},

	73:	{
		SkillPoints: [-1],
		StatPoints: [-1, -1, -1, -1, -1],
		Update: function () {
			Config.AttackSkill = [-1, 49, 42, 53, 42, -1, -1];


		}
	},

	74:	{
		SkillPoints: [-1],
		StatPoints: [-1, -1, -1, -1, -1],
		Update: function () {
			Config.AttackSkill = [-1, 49, 42, 53, 42, -1, -1];


		}
	},

	75:	{
		SkillPoints: [-1],
		StatPoints: [-1, -1, -1, -1, -1],
		Update: function () {
			Config.AttackSkill = [-1, 49, 42, 53, 42, -1, -1];


		}
	},

	76:	{
		SkillPoints: [-1],
		StatPoints: [-1, -1, -1, -1, -1],
		Update: function () {
			Config.AttackSkill = [-1, 49, 42, 53, 42, -1, -1];


		}
	},

	77:	{
		SkillPoints: [-1],
		StatPoints: [-1, -1, -1, -1, -1],
		Update: function () {
			Config.AttackSkill = [-1, 49, 42, 53, 42, -1, -1];


		}
	},

	78:	{
		SkillPoints: [-1],
		StatPoints: [-1, -1, -1, -1, -1],
		Update: function () {
			Config.AttackSkill = [-1, 49, 42, 53, 42, -1, -1];


		}
	},

	79:	{
		SkillPoints: [-1],
		StatPoints: [-1, -1, -1, -1, -1],
		Update: function () {
			Config.AttackSkill = [-1, 49, 42, 53, 42, -1, -1];


		}
	},

	80:	{
		SkillPoints: [-1],
		StatPoints: [-1, -1, -1, -1, -1],
		Update: function () {
			Config.AttackSkill = [-1, 49, 42, 53, 42, -1, -1];


		}
	},

	81:	{
		SkillPoints: [-1],
		StatPoints: [-1, -1, -1, -1, -1],
		Update: function () {
			Config.AttackSkill = [-1, 49, 42, 53, 42, -1, -1];


		}
	},

	82:	{
		SkillPoints: [-1],
		StatPoints: [-1, -1, -1, -1, -1],
		Update: function () {
			Config.AttackSkill = [-1, 49, 42, 53, 42, -1, -1];


		}
	},

	83:	{
		SkillPoints: [-1],
		StatPoints: [-1, -1, -1, -1, -1],
		Update: function () {
			Config.AttackSkill = [-1, 49, 42, 53, 42, -1, -1];


		}
	},

	84:	{
		SkillPoints: [-1],
		StatPoints: [-1, -1, -1, -1, -1],
		Update: function () {
			Config.AttackSkill = [-1, 49, 42, 53, 42, -1, -1];


		}
	},

	85:	{
		SkillPoints: [-1],
		StatPoints: [-1, -1, -1, -1, -1],
		Update: function () {
			Config.AttackSkill = [-1, 49, 42, 53, 42, -1, -1];


		}
	},

	86:	{
		SkillPoints: [-1],
		StatPoints: [-1, -1, -1, -1, -1],
		Update: function () {
			Config.AttackSkill = [-1, 49, 42, 53, 42, -1, -1];


		}
	},

	87:	{
		SkillPoints: [-1],
		StatPoints: [-1, -1, -1, -1, -1],
		Update: function () {
			Config.AttackSkill = [-1, 49, 42, 53, 42, -1, -1];


		}
	},

	88:	{
		SkillPoints: [-1],
		StatPoints: [-1, -1, -1, -1, -1],
		Update: function () {
			Config.AttackSkill = [-1, 49, 42, 53, 42, -1, -1];


		}
	},

	89:	{
		SkillPoints: [-1],
		StatPoints: [-1, -1, -1, -1, -1],
		Update: function () {
			Config.AttackSkill = [-1, 49, 42, 53, 42, -1, -1];


		}
	},

	90:	{
		SkillPoints: [-1],
		StatPoints: [-1, -1, -1, -1, -1],
		Update: function () {
			Config.AttackSkill = [-1, 49, 42, 53, 42, -1, -1];


		}
	},

	91:	{
		SkillPoints: [-1],
		StatPoints: [-1, -1, -1, -1, -1],
		Update: function () {
			Config.AttackSkill = [-1, 49, 42, 53, 42, -1, -1];


		}
	},

	92:	{
		SkillPoints: [-1],
		StatPoints: [-1, -1, -1, -1, -1],
		Update: function () {
			Config.AttackSkill = [-1, 49, 42, 53, 42, -1, -1];


		}
	},

	93:	{
		SkillPoints: [-1],
		StatPoints: [-1, -1, -1, -1, -1],
		Update: function () {
			Config.AttackSkill = [-1, 49, 42, 53, 42, -1, -1];


		}
	},

	94:	{
		SkillPoints: [-1],
		StatPoints: [-1, -1, -1, -1, -1],
		Update: function () {
			Config.AttackSkill = [-1, 49, 42, 53, 42, -1, -1];


		}
	},

	95:	{
		SkillPoints: [-1],
		StatPoints: [-1, -1, -1, -1, -1],
		Update: function () {
			Config.AttackSkill = [-1, 49, 42, 53, 42, -1, -1];


		}
	},

	96:	{
		SkillPoints: [-1],
		StatPoints: [-1, -1, -1, -1, -1],
		Update: function () {
			Config.AttackSkill = [-1, 49, 42, 53, 42, -1, -1];


		}
	},

	97:	{
		SkillPoints: [-1],
		StatPoints: [-1, -1, -1, -1, -1],
		Update: function () {
			Config.AttackSkill = [-1, 49, 42, 53, 42, -1, -1];


		}
	},

	98:	{
		SkillPoints: [-1],
		StatPoints: [-1, -1, -1, -1, -1],
		Update: function () {
			Config.AttackSkill = [-1, 49, 42, 53, 42, -1, -1];


		}
	},

	99:	{
		SkillPoints: [-1],
		StatPoints: [-1, -1, -1, -1, -1],
		Update: function () {
			Config.AttackSkill = [-1, 49, 42, 53, 42, -1, -1];


		}
	}
};

**Reply by zylo-** [reply]
> Yes, with the kind assistance from some users of my script, we are working towards optimizing the leveling builds as well as changing the way the finalbuild respec is called. This way the bot will be able to respec to a light sorc after having infinity, instead of at defined character level.

This would be amazing. Exactly what I am looking for. 
I would tip well for a script that

- lvls blizzballer then switches to lightning build when it can build infinity for merc
- builds spirit monarch and equips it (any)  and keeps building until it gets 35 fcr one
- puts spirit monarch and CTA on the switch
- allows me to specify the full build item path (for example i'd rather tals armor belt ammy only, and occy and shako, ptravs and spirit monarch, pchancies, nagels until 40mf magic rings ) - i'd like to be able to specify in length the build path, for example, use tals belt 10% mf, unless 11%, ... unless 15%
- sockets occy and shako and puts ist in it
- sockets merc andy's and puts ral in it
- merc uses fort or CoH or treachery or shaftstop
- merc uses andys, vampire gaze, tals helm
- keeps a few 7% mf scs
- keeps&uses 40% mf gheeds?

**Reply by ddarbs** [reply]
With what I have in my comment above plus a few changes to globals you can have it level blizzballer then swap to lightning once it has infinity. I think the current SoloLeveling already builds spirit monarchs and CTA for swaps, and as far as specifying build item path: in the Sorceress.SoloLeveling.js I believe you can look under var autoequipTiers = [ to change what build order you want with the [tier]. Same for var autoequipmercTiers = [. The charms I don't have an answer for you, but they said they're working on that.

**Reply by isid0re** [reply]
I'm already moving in thst direction. The final build will respec for light sorc once  has infinity that is already implemented. 

As for the build path you are looking for. 
Just add them to the autoequiptiers variable in the sorceress.config. 

Charms are a bit tricky right now. And gheeds is hard to upgrade because you can only hold one at a time. So it's a matter of dropping store value  check new then drop new if less and pick new. But doing that u run a risk of losing both 

Cta and spirit are already there for switch. Just needs str for monarch

So basically the only 2 things my bot csnt do for you right now are socketing gear (larzuk quest- in process) and charms (in process). 

So dont worry it will come. This isnt a team working in this. It is just 1 mans covid hobby.
